anyway/infographics_utils.py

```diff
@@ -1118,22 +1120,53 @@ def localize_items(request_params: RequestParams, items: Dict) -> Dict:
+# TODO: As it currently stands, for SubUrbanWidgets such as this, the is_relevant() method is simply is_sub_urban() which checks for
+# the presence of "road1" and "road_segment_name", which do not match the condition of "street1/2_hebrew" having value, which is required
+#  for matching with the "involved_markers_hebrew" table (see SQLAlchemy below). Maybe we should make InjuredAccidentsWithPedestriansWidget
+# inherit Widget instead of SubUrbanWidget?
 class InjuredAccidentsWithPedestriansWidget(SubUrbanWidget):
     name: str = "injured_accidents_with_pedestrians"
 
+    def validate_parameters(self):
+        # TODO: validate each parameter and display message accordingly
+        return self.request_params.location_info.get('yishuv_name') is not None and \
+            (self.request_params.news_flash_obj.street1_hebrew is not None or   \
+            self.request_params.news_flash_obj.street2_hebrew is not None) and  \
+            self.request_params.years_ago is None
+
     def __init__(self, request_params: RequestParams):
         super().__init__(request_params, type(self).name)
+
+        if not self.validate_parameters():
+            logging.exception(f"Could not validate parameters for {NewsFlash} : {request_params.news_flash_obj.id}")
+            return None
+
         self.rank = 18
-        self.text = {"title": "נפגעים הולכי רגל ברחוב ז׳בוטינסקי, פתח תקווה"}
+        self.text = {"title": f"נפגעים הולכי רגל ברחוב(ות) {str.join(', ', [request_params.news_flash_obj.street1_hebrew, request_params.news_flash_obj.street2_hebrew])}, {request_params.news_flash_obj.yishuv_name}"}
 
     @staticmethod
+    # TODO: change?
     def is_in_cache() -> bool:
         return False
 
     def generate_items(self) -> None:
-        self.items = (
-            InjuredAccidentsWithPedestriansWidget.injured_accidents_with_pedestrians_mock_data()
-        )
+        try:
+            query = db.session.query(InvolvedMarkerView)\
+                .with_entities(InvolvedMarkerView.accident_year,    \
+                    InvolvedMarkerView.injury_severity, \
+                    InvolvedMarkerView.injury_severity_hebrew,\
+                    func.count(distinct(f"{InvolvedMarkerView.provider_and_id}_{InvolvedMarkerView.involve_id}")))  \
+                .filter(InvolvedMarkerView.accident_yishuv_name == self.request_params.location_info['yishuv_name'])\
+                .filter(InvolvedMarkerView.injured_type == 1)   \
+                .filter(or_(InvolvedMarkerView.street1_hebrew == self.request_params.news_flash_obj.street1_hebrew,
+                    InvolvedMarkerView.street2_hebrew == self.request_params.news_flash_obj.street2_hebrew))    \
+                .filter(InvolvedMarkerView.accident_year >= datetime.date.today().year - self.request_params.years_ago)   \
+                .group_by(InvolvedMarkerView.accident_year, InvolvedMarkerView.injury_severity, InvolvedMarkerView.injury_severity_hebrew)
+
+            self.items = (query.all())
```

Comment:
I suggest building items in a similar way to the way @ziv17 built his 2 recent stacked bar charts ([see this pr](https://github.com/hasadna/anyway/pull/1875))

Reply:
do you mean using "get_accidents_stats"?

Reply:
No. You can use get_accidents_stats - but as you mentioned above - it's not suitable for this specific use case since there is no or_/and_ clauses of SQLAlchemy in this function.

I mean the final widget dictionary that we output - I think we need to be consistent on this one. 
@ziv17 built a structure for stacked bar charts and I think we should use it for all bar charts we create as widgets - including this one.

Reply:
I understand from the PR that we'd like to have a 2-level dict as output with the first level as "injury_severity" and the 2nd level as "accident_year".
in that case i have a few questions:
1. in the redash there's grouping by injury_severity_hebrew, which theoretically adds another level. what should we do about it? just remove it?
2. is this 2-level something that is fixed? e.g. will it always be a 2-level dict by severity/year? if not, how do we generalize?

Reply:
![image](https://user-images.githubusercontent.com/24354642/133788423-bb14013c-ea8b-4afd-9f28-29b193b74656.png)